dfget/core/api/supernode_api.go

```diff
@@ -100,9 +103,13 @@ func (api *supernodeAPI) ReportPiece(node string, req *types.ReportPieceRequest)
 
 	url := fmt.Sprintf("%s://%s%s?%s",
 		api.Scheme, node, peerReportPiecePath, util.ParseQuery(req))
-
 	resp = new(types.BaseResponse)
 	e = api.get(url, resp)
+	if e != nil {
+		logrus.Infof("failed to report piece{taskid:%s,range:%s},err: %v", req.TaskID, req.PieceRange, e)
+	} else if resp != nil && resp.Code != constants.CodeGetPieceReport {
```

Comment:
We should record the info about **resp** to make it more valuable.

Reply:
but I have put the resp.code into log already.

Reply:
> We should record the info about **resp** to make it more valuable.
but I have put the resp.code into log already.

Reply:
but I have put the resp.code into log already.

Reply:
If there is a `return` of the if block, there is no need to add `else`. The following is better:
```
if e != nil {
    logrus.Errorf("blabla")
    return nil, e
}

if resp != nil && resp.Code != constants.CodeGetPieceReport {
    
}
```

Reply:
Done